spec.bs

```diff
@@ -185,44 +185,59 @@ Each {{URLPattern}} object has an associated <dfn for=URLPattern>search componen
 Each {{URLPattern}} object has an associated <dfn for=URLPattern>hash component</dfn>, a [=component=], which must be set upon creation.
 
 <dl class="domintro non-normative">
-  <dt><code>{{URLPattern}} . {{URLPattern/protocol}}</code>
+  <dt><code>|urlPattern| = new {{URLPattern/constructor(input, baseURL)|URLPattern}}(|input|, |baseURL|)</code></dt>
+  <dd class=allow-2119>
+    Constructs a new URLPattern object.  The |input| may be a {{URLPatternInit}} dictionary or a string.  If |input| is a string, then an optional |baseURL| string may also be provided.
+  </dd>
+
+  <dt><code>|matches| = |urlPattern|.{{URLPattern/test(input, baseURL)|test}}(|input|, |baseURL|)</code></dt>
+  <dd class=allow-2119>
+    Tests if |urlPattern| matches the given arguments.  The |input| may be a {{URLPatternInit}} dictionary or a string.  If |input| is a string, then an optional |baseURL| string may also be provided.  The returned |matches| value is a boolean.
+  </dd>
+
+  <dt><code>|result| = |urlPattern|.{{URLPattern/exec(input, baseURL)|exec}}(|input|, |baseURL|)</code></dt>
+  <dd class=allow-2119>
+    Executes the |urlPattern| against the given arguments.  The |input| may be a {{URLPatternInit}} dictionary or a string.  If |input| is a string, then an optional |baseURL| string may also be provided.  If the arguments did not match the |urlPattern|, then |result| will be null.  If the arguments did match, then |result| will be a {{URLPatternResult}} providing details about values that matched groups within |urlPattern|.
```

Comment:
Similarly describing |result| in more detail than just referring to URLPatternResult would be ideal. https://html.spec.whatwg.org/#2dcontext:dom-context-2d-canvas-getcontextattributes-2 is an example of what we've done for return values in the past.

Reply:
I tried to describe it better, but did not break it down like the example.  I ran out of time to try that today.  Also, since its got multiple nested objects, not sure if that would be very clear.